### PR TITLE
754 full end to end test expanded to include routing on radio and number answers 4

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
This Pr is part of the setup work for the routing refactor that will be taking place this sprint. It add a end-2-end test that will test the ui, publisher and the runner implementation of routing and help us to ensure that all functionality is preserved after this routing refactor.

### How to review 
Test looks comprehensive and all passes.

### Test is currently broken due to a bug that is being fixed in: https://github.com/ONSdigital/eq-author-app/pull/53

### CODE REVIEW

File: eq-author/cypress/routingTests/routing_e2e_readme.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Routing E2E Test
+
+This test spec file was created as part of the setup work for the routing refactor. It is designed to create a simple survey that utilizes all
```
Comment (adammahmood) [inline eq-author/cypress/routingTests/routing_e2e_readme.md]: Do we plan on keeping these tests after the refactor?

File: eq-author/cypress/builders/answer.js [hunk 2]
```diff
@@ -1,23 +1,50 @@
...
   });
 };
 
+const configueBasicAnswer = ({ label, description }) => {
+  if (label) {
+    cy.get(testId("txt-answer-label")).type(label);
+  }
+  if (description) {
+    cy.get(testId("txt-answer-description")).then(
+      ele =>
```
Comment (adammahmood) [inline eq-author/cypress/builders/answer.js]: Why `ele`?

File: eq-author/cypress/builders/answer.js [hunk 3]
```diff
@@ -1,23 +1,50 @@
 import { testId } from "../utils";
 
-const configureCheckbox = ({ options }) => {
+const configureMultipleChoice = initialCount => ({ options }) => {
```
Comment (adammahmood) [inline eq-author/cypress/builders/answer.js]: Why not pass this as an object shaped as `{initialCount, options}` ?

File: eq-author/cypress/routingTests/routing_e2e_spec.js [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+      }
+    });
+  });
+
+  it("adds radio routing", () => {
+    navigateToPage("Section 1, Page 1");
+    navigateToRoutingTab();
+    cy.get(testId("btn-add-rule")).click();
+
+    cy.get(testId("routing-editor")).within(() => {
```
Comment (adammahmood) [inline eq-author/cypress/routingTests/routing_e2e_spec.js]: Some of these tests are shared with `routing_spec.js`.  Can we share them?

File: eq-author/cypress/routingTests/routing_e2e_spec.js [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+          cy.get(testId("comparator-selector")).select("GreaterThan");
+          cy.get(testId("number-input")).type("5");
+          findByLabel("THEN")
+            .last()
+            .select("End of Questionnaire");
+        });
+    });
+  });
+
+  describe("Runner Side", () => {
```
Comment (adammahmood) [inline eq-author/cypress/routingTests/routing_e2e_spec.js]: Can we rename this to something more descriptive? Maybe just `Runner`?

File: eq-author/cypress/routingTests/routing_e2e_readme.md (hunk 1 above)
Comment (SamGodwin2) [reply eq-author/cypress/routingTests/routing_e2e_readme.md]: I wasn't planning on it as they're unlikely to be very reliable as they depend on Runner which could change without our knowledge?

File: eq-author/package.json [hunk 6]
```diff
@@ -23,7 +24,8 @@
     "precommit": "lint-staged",
     "coverage": "yarn test -- --coverage",
     "cypress:open": "CYPRESS_baseUrl=${CYPRESS_baseUrl:=http://localhost:3000/} cypress open",
-    "cypress:open:e2e": "CYPRESS_baseUrl=${CYPRESS_baseUrl:=http://localhost:13000/} cypress open --config integrationFolder=cypress/e2e"
+    "cypress:open:e2e": "CYPRESS_baseUrl=${CYPRESS_baseUrl:=http://localhost:13000/} cypress open --config integrationFolder=cypress/e2e",
+    "cypress:open:routing": "CYPRESS_baseUrl=${CYPRESS_baseUrl:=http://localhost:3000/} CYPRESS_chromeWebSecurity='false' cypress open --config integrationFolder=cypress/routingTests"
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/package.json]: It also relies on having a local version of the app/api/publisher running as well, and we don't spin them up in the other cypress tests so as long as it's detailed in the readme what you need running it seems okay? I'll double check the readme and make sure that's clear though.

File: eq-author/cypress/routingTests/routing_e2e_spec.js (hunk 4 above)
Comment (SamGodwin2) [reply eq-author/cypress/routingTests/routing_e2e_spec.js]: I didn't really want to share anything here except for the utils as we wanted these tests to be very specific and isolated from the other routing implementation.

**Comment by SamGodwin2** [comment]
@adammahmood think the issue with the runner side is due to the api not being spun up using `RUNNER_SESSION_URL="http://localhost:5000/session?token="` as for some reason cypress doesn't play nice with `docker.for.mac`?

File: eq-author/cypress/builders/answer.js (hunk 2 above)
Comment (SamGodwin2) [reply eq-author/cypress/builders/answer.js]: Just the name I gave to the Jquery element the `cy.get` returns, got a clearer name?

File: eq-author/cypress/routingTests/routing_e2e_spec.js (hunk 4 above)
Comment (tgandrews) [reply eq-author/cypress/routingTests/routing_e2e_spec.js]: We could move the actions to builders? So we can reuse the same functionality in both places

File: eq-author/cypress/routingTests/routing_e2e_readme.md (hunk 1 above)
Comment (tgandrews) [reply eq-author/cypress/routingTests/routing_e2e_readme.md]: Although the team restructure may be stopping that as runner is code frozen?

File: eq-author/cypress/builders/answer.js (hunk 2 above)
Comment (tgandrews) [reply eq-author/cypress/builders/answer.js]: `$el` was the jQuery convention IIRC

File: eq-author/cypress/routingTests/routing_e2e_readme.md (hunk 1 above)
Comment (tgandrews) [reply eq-author/cypress/routingTests/routing_e2e_readme.md]: The publisher output sounds like a good idea once we have proven the refactor works.

File: eq-author/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (tgandrews) [reply eq-author/package.json]: The cypress dockerfile includes publisher and validator which is probably fine as thats the limit or our remit. But including runner would seem like too much.

**Pull request merged**